src/utils/config_filters.cpp

```diff
@@ -62,8 +62,10 @@ bool utils::config_filters::int_matches_if_present(const config& filter, const c
 		return false;
 	}
 
-	int value_def = def ? (*def) : 0;
-	return in_ranges<int>(cfg[attribute].to_int(value_def), utils::parse_ranges_int(filter[attribute].str()));
+	if(def){
+		return in_ranges<int>(cfg[attribute].to_int(*def), utils::parse_ranges_int(filter[attribute].str()));
+	}
+	return in_ranges<int>(cfg[attribute].to_int(), utils::parse_ranges_int(filter[attribute].str()));
```

Comment:
I understand the changes in the .hpp file, but why are these changes in the .cpp file needed?

Reply:
because define default_value nedded if not have attribute but avec def defined, in last line attribute is present but not def

Reply:
I believe the old and the new code in the .cpp get the same result:
* if `cfg` has that attribute, always compare to the value in `cfg`
* if `cfg` doesn't have the attribute but `def` is not `nullopt`, compare to `def`
* otherwise `return false` before reaching the changed lines

If I've missed something, please give an example of inputs which would generate a different output.

(Note: only asking about the .cpp changes here. The .hpp changes look good.)

Reply:
Here it don't have default vale anymore when def is nullopt then before it have same if def not defined and has cfg attribute,value who unusable

Reply:
But if `cfg` has an unusable value then `to_int()` will itself fall back to the default argument in config_attribute_value.hpp:
```
	int to_int(int def = 0) const;
```

So I think it ends up with the same result as the old .cpp code.

Reply:
I mean, if someone does something like `multiply=h`, then it doesn't seem unexpected that they'd get an odd result.

Reply:
reverted remove value_def code